src/main/java/org/springframework/samples/petclinic/owner/PetController.java

```diff
@@ -107,6 +112,18 @@ public String initUpdateForm(Owner owner, @PathVariable("petId") int petId, Mode
 
 	@PostMapping("/pets/{petId}/edit")
 	public String processUpdateForm(@Valid Pet pet, BindingResult result, Owner owner, ModelMap model) {
+
+		if (StringUtils.hasLength(pet.getName()) && pet.isNew() && owner.getPet(pet.getName(), true) != null) {
```

Comment:
Is this necessary - it's not a new pet, so the test will always fail.

Reply:
you are correct this was not there in the latest version .
sorry i maybe i added it by mistake 
its fixed now in the new commit